CONTRIBUTING.md

```diff
@@ -1,15 +1,109 @@
 # Contributing
 
-Before contributing, please read [our code style](https://github.com/teloxide/teloxide/blob/master/CODE_STYLE.md) and [the license](https://github.com/teloxide/teloxide/blob/master/LICENSE).
+Before contributing, please read [our code style](./CODE_STYLE.md) and [the license](./LICENSE).
 
-To change the source code, fork the `master` branch of this repository and work inside your own branch. Then send us a PR into `master` branch and wait for the CI to check everything. However, you'd better check changes first locally:
+> **Note**
+>
+> These contributing instructions might not be fully up-to-date or complete.
+> However, they should be a good starting point.
+>
+> If you find inaccuracies / missing things, please expand this or contact us.
 
+## Reporting bugs, questions, feature requests
+
+To report a bug or suggest new functionality go to [issues](https://github.com/teloxide/teloxide/issues).
+Try to make MRE (**M**inimal **R**eproducible **E**xample) and specify your teloxide version to let others help you.
+
+If you want to ask a question, you can either
+- Open a new [github discussion](https://github.com/teloxide/teloxide/discussions), or
+- Write to our telegram group ([eng](https://t.me/teloxide), [ru](https://t.me/teloxide_ru))
+
+## Code
+
+### Git
+
+To change the source code, you need a local copy of it.
+Fork the `master` branch of this repository via github and clone your fork locally.
+
+When working on a new thing, create a new branch with `git switch -c my-branch-name` (or other commands that work with branches).
+This way it will be easier to manage when you want to do other things.
+
+When your changes are ready, you can open a github pull request.
+
+If your branch has conflicts with master please resolve them by doing something like this:
+```shell
+# Temporary switch to master branch
+git switch master
+
+# Pull changes from the upstream.
+# You may need to use something different from "origin",
+# depending on how you setup your remotes.
+git pull origin master
+
+# Switch back to your feature branch
+git switch -
+
+# Move your changes on top of changes in master branch.
+git rebase master
+
+# Here you'll need to resolve the conflicts,
+# git commands will print some guidance.
+
+# Once conflicts are resolved,
+# forcefully push the changes to your fork
+git push --force-with-lease
 ```
-cargo clippy --all --all-features --all-targets
```

Comment:
Here we might suggest the preferred way of structuring pull requests:

### Pull request structure
> If your pull request solves on of  [issues](https://github.com/teloxide/teloxide/issues) don't forget to specify it's number (i.e. #nnn)
> If your pull request suggests new functionality or some changes, explain your point of view in the next form:
```
# Pros
- ...

# Cons
- ...

# Alternatives
if any
```

Reply:
@WaffleLapkin, is this resolved?

Reply:
@Hirrolot I'm not sure this is a good idea, idk. I feel like rigid structure discourages contributions, because they often don't fit the structure. You don't always have enough to write a list in all three of those sections.

I've added some guidance in c934e678f6a9cbf0e367505b3fd1eb53ec358376, does it resolve this?

Reply:
I agree that PR summaries often don't fit in the rigid structure.